src/DIPS.Xamarin.UI/Controls/Sheet/SheetBehavior.cs

```diff
@@ -251,8 +284,230 @@ private static void BackgroundColorPropertyChanged(BindableObject bindable, obje
+            nameof(IsCancelButtonVisible),
+            typeof(bool),
+            typeof(SheetBehavior),
+            false);
+
+        /// <summary>
+        /// <see cref="CancelCommand"/>
+        /// </summary>
+        public static readonly BindableProperty CancelCommandProperty = BindableProperty.Create(
+            nameof(CancelCommand),
+            typeof(ICancelSheetCommand),
+            typeof(SheetBehavior),
+            defaultValueCreator: DefaultValueCreator);
+
+        /// <summary>
+        ///     <see cref="CancelCommandParameter" />
+        /// </summary>
+        public static readonly BindableProperty CancelCommandParameterProperty =
+            BindableProperty.Create(nameof(CancelCommandParameter), typeof(object), typeof(SheetBehavior));
+
+        /// <summary>
+        ///     <see cref="ActionCommand" />
+        /// </summary>
+        public static readonly BindableProperty ActionCommandProperty = BindableProperty.Create(
+            nameof(ActionCommand),
+            typeof(ICommand),
+            typeof(SheetBehavior));
+
+        /// <summary>
+        ///     <see cref="ActionCommandParameter" />
+        /// </summary>
+        public static readonly BindableProperty ActionCommandParameterProperty =
+            BindableProperty.Create(nameof(ActionCommandParameter), typeof(object), typeof(SheetBehavior));
+
+        private readonly double m_autoCloseThreshold = 0.05;
+
+        private bool m_fromIsOpenContext;
+        private ModalityLayout? m_modalityLayout;
+
+        private double m_originalPosition = -1;
+
+        private SheetView? m_sheetView;
+
+        /// <summary>
+        ///     Parameter passed to <see cref="ActionCommand" />.
+        ///     This is a bindable property.
+        /// </summary>
+        public object ActionCommandParameter
+        {
+            get => GetValue(ActionCommandParameterProperty);
+            set => SetValue(ActionCommandParameterProperty, value);
+        }
+
+        /// <summary>
+        ///     Gets or sets the command that is invoked when the action button is activated.
+        ///     This is a bindable property.
+        /// </summary>
+        public ICommand ActionCommand
+        {
+            get => (ICommand)GetValue(ActionCommandProperty);
+            set => SetValue(ActionCommandProperty, value);
+        }
+
+        /// <summary>
+        ///     Parameter passed to <see cref="CancelCommand" />.
+        ///     This is a bindable property.
+        /// </summary>
+        public object CancelCommandParameter
+        {
+            get => GetValue(CancelCommandParameterProperty);
+            set => SetValue(CancelCommandParameterProperty, value);
+        }
+
+        /// <summary>
+        ///     Gets or sets the command that is invoked when the cancel button is activated.
+        ///     This is a bindable property.
+        /// </summary>
+        public ICancelSheetCommand CancelCommand

```

Comment:
One drawback of this being `ICancelSheetCommand` is that new consumers that don't know any better will most likely create an `ICommand` the first time they test this without reading the docs.
What happens then is that their Command won't get executed as the binding does not match. A

One kind of nice way to notify the consumer that they have set the wrong type of command is to use a `TypeConverter` and throw if the binding is not the correct type. Would you mind using a couple of minutes on that and see if it works, if it takes to much time w e an do it later.

There might be other ways to do it aswell though.

Reply:
One a second thought, not sure if it will work. But we can try.

Reply:
Can change to be of type `ICommand`? Close sheet if its set to a `Command`.

Reply:
And still handle if its a `CloseSheetCommand`

Reply:
That's interesting and I think its a good solution. Thoughts @Illedan?